sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExtractValue.scala

```diff
@@ -195,6 +205,10 @@ case class GetArrayItem(child: Expression, ordinal: Expression)
       baseValue(index)
     }
   }
+
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    defineCodeGen(ctx, ev, (eval1, eval2) => s"(${ctx.javaType(dataType)})$eval1.apply((int)$eval2)")
```

Comment:
It's weird that I have to add type cast here. Why `scala.collection.Seq<T>.apply` did't return `T` but `Object`? I test it locally, it do return `T`, but in code gen, it returns `Object`.